### PR TITLE
:bug: Fix code snippet viewer for issues and insights

### LINKED ISSUE #2495: Improve regex pattern documentation and robustness in incident-code-snip-viewer
## Description

This issue tracks improvements needed for the regex pattern used in the incident code snippet viewer component.

## Context

This issue was identified during code review of PR #2488 (Create Insights page) in the following file:
`client/src/app/pages/insights/insight-detail-drawer/file-incidents-detail-modal/incident-code-snip-viewer.tsx`

### 1. Regex Pattern Documentation and Improvement

**Current code:**
```typescript
const codeLineRegex = /^\s*([0-9]+)( {2})?(.*)$/; // Pattern: leading whitespace (line number) (2 spaces)? (code)
```

**Improvements needed:**
- Add comprehensive documentation explaining the expected input format
- Use `\d+` instead of `[0-9]+` for better specificity and clarity
- Consider making the pattern more robust for edge cases

**Suggested improvement:**
```typescript
// Matches lines in format: "  123  actual code content" or "  123actual code content"
// Group 1: line number, Group 2: optional 2 spaces, Group 3: code content
const codeLineRegex = /^\s*(\d+)(\s{2})?(.*)$/;
```

## Additional Considerations

- Validate the regex against expected input formats to ensure robustness
- Consider adding unit tests for the regex pattern parsing logic
- Review if the pattern handles all edge cases in real-world code snippets

## References

- Original PR: https://github.com/konveyor/tackle2-ui/pull/2488
- Review comment: https://github.com/konveyor/tackle2-ui/pull/2488#discussion_r2225731056
- Requested by: @sjd78

## Acceptance Criteria

- [ ] Regex pattern has comprehensive documentation
- [ ] Pattern uses \d+ instead of [0-9]+
- [ ] Pattern is validated against expected input formats
- [ ] Unit tests are added for the parsing logic (optional but recommended)

### LINKED ISSUE #2849: [BUG] Analysis results for all applications not rendering proper code snippets
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

v0.9.0-alpha.4

### Priority

Blocker

### Current Behavior

Run analysis with openjDK target (or any target). Check an incident that's line number is beyond the default 21 lines that get rendered. The code snippet is not formatted correctly and the offending line is not rendered.

Image

### Expected Behavior

Code snippet shows expected output

### How Reproducible

Always (Default)

### Steps To Reproduce

See above
### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sjd78**
The code snippet viewer was not correctly interpreting the code snippets for display in the `CodeEditor` for issues and insights.

  - Updated the codeSnip parsing to properly handle the codeSnip lines.
  - Made the code snippet viewer a common component
  - Fixed the incident marker display
  - Fixed the icon display

Fixes: #2849
Fixes: #2495

AI Attribution: AIA Human-AI blend, Human-initiated, Reviewed, claude-4.5-opus-high v1.0

## Summary by CodeRabbit

* **New Features**
  * Unified code snippet viewer shows a centered exclamation error indicator on incident lines and displays contextual marker messages.

* **Improvements**
  * More reliable parsing of numbered code snippets for accurate line mapping and precise incident highlighting.
  * Viewer usage consolidated across pages for consistent behavior and props.

* **Tests**
  * Added unit tests covering code-snippet parsing.

### CODE REVIEW

**Review by rszwajko — CHANGES_REQUESTED** [review]
1. The refactoring (removing the duplication) is definitely needed here however it hides the actual fix. Please highlight the fix is some way i.e. by splitting into 2 commits (refactoring +fix).
2. It's not clear to me how much the text can be malformed - do we need a more defensive approach (i.e. assume the text can be random) or perhaps it's bug in some other component and the text should be guaranteed to be valid.
3. unit tests for the `parseCodeSnip` (with example malformed text) would help to understand the code

File: client/src/app/components/code-snip-viewer/incident-code-snip-viewer.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  for (const numberedLine of numberedLines) {
+    const match = numberedLine.match(codeLineRegex);
+    if (match) {
+      const lineNum = Number(match[1]);
+      if (!startLineFound && !isNaN(lineNum)) {
+        startLine = lineNum;
+        startLineFound = true;
+      }
+      // match[3] is the code content after the line number and optional separator
+      codeLines.push(match[3] ?? "");
```
[the remark this answers is not in the document]
Comment (sjd78) [reply client/src/app/components/code-snip-viewer/incident-code-snip-viewer.tsx]: Also moved the parse code to `client/src/app/utils/code-snip-utils.ts`

File: client/src/app/components/code-snip-viewer/incident-code-snip-viewer.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+          {t("message.noCodesSnippetAvailableBody")}
+        </EmptyStateBody>
+      </EmptyState>
+    );
+  }
+
+  const { startLine, code, lineCount } = parseCodeSnip(incident.codeSnip);
+
+  // Convert absolute file line numbers to/from relative editor line numbers
+  const toRelativeLine = (absoluteLine: number) => absoluteLine - startLine + 1;
```
[the remark this answers is not in the document]
Comment (sjd78) [reply client/src/app/components/code-snip-viewer/incident-code-snip-viewer.tsx]: Renaming them will help.  They just translate between the line number in the code snip and the line number displayed in the code editor.

**Review by rszwajko — APPROVED** [review]
